etcdserver/backend.go

```diff
@@ -96,9 +94,7 @@ func openBackend(cfg ServerConfig) backend.Backend {
 // case, replace the db with the snapshot db sent by the leader.
 func recoverSnapshotBackend(cfg ServerConfig, oldbe backend.Backend, snapshot raftpb.Snapshot) (backend.Backend, error) {
 	ci := cindex.NewConsistentIndex(oldbe.BatchTx())
-	kv := mvcc.New(cfg.Logger, oldbe, &lease.FakeLessor{}, ci, mvcc.StoreConfig{CompactionBatchLimit: cfg.CompactionBatchLimit})
-	defer kv.Close()
-	if snapshot.Metadata.Index <= kv.ConsistentIndex() {
+	if snapshot.Metadata.Index <= ci.ConsistentIndex() {
```

Comment:
we have to make sure the backend has consistentIndexKeyName key，Otherwise, boltdb will throw an exception. recoverSnapshotBackend  is only called  when wal exists. I think it is safe.  how do you think so? @gyuho

Reply:
Hmm, this is called before we call `mvcc.NewStore`...

Reply:
Can we do something like

```go
	if beExist {
		kvindex := srv.kv.ConsistentIndex()
```

Reply:
Is there any exception that wal file exists but db does not？

Reply:
> Can we do something like
> 
> ```go
> 	if beExist {
> 		kvindex := srv.kv.ConsistentIndex()
> ```

done,it will  make recoverSnapshotBackend more robust,thanks.

Reply:
> Is there any exception that wal file exists but db does not？

I don't think so, unless user sets up a separate WAL dir, and deletes data dir intentionally.